cmd/build/v2/services.go

```diff
@@ -118,3 +118,24 @@ func (bc *OktetoBuilder) checkImageAtGlobalAndSetEnvs(service string, options *t
 	return true, nil
 
 }
+
+// GetServicesToBuildFromSubset returns the services it has to built because they are not already built from a subset of services
+func (bc *OktetoBuilder) GetServicesToBuildFromSubset(ctx context.Context, manifest *model.Manifest, subset []string) ([]string, error) {
+	for name := range manifest.Build {
+		needsBuild := false
+		for _, toBuildName := range subset {
+			if name == toBuildName {
+				needsBuild = true
+				break
+			}
+		}
+		if !needsBuild {
```

Comment:
I think you don't need this check. In the only place where the flag is set to `true` you are breaking the loop, so if the code arrives to this point, this flag is `false`

Reply:
This break statement breaks the inner loop because we need to check if the services that needs to be built are in the manifest.Build section so we need two for statements

Reply:
oh, yes, there were two nested loops, I didn't see it well, thanks for the explanation